src/main/java/hudson/plugins/jdepend/JDependBuildAction.java

```diff
@@ -8,23 +8,35 @@
 
 import hudson.model.AbstractBuild;
 import hudson.model.Action;
+import hudson.model.Run;
+import jenkins.model.RunAction2;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
 
 /**
  * A build action to generate JDepend HTML reports
  * @author cflewis
  *
  */
-public class JDependBuildAction implements Action 
+public class JDependBuildAction implements RunAction2
 {
-	public final AbstractBuild<?, ?> build;
-	private final JDependParser jDependParser;
```

Comment:
IIUC this is the actual fix—the plugin was for some reason saving an instance field for an object used only in the constructor? Sigh.